### ISSUE TITLE
hacker news feeds not updating correctly

### ISSUE THREAD

**Issue by rattasak** [samuelclay/NewsBlur#916]
This might very well be user error on my part, but trying to pull from rss feeds for hacker news based on number of points seem to not be working. When using the feed from hnrss (filter for posts with greater than 75 points), I don't get any new posts at all (most recent that shows up is from yesterday). It seems like it might be some kind of caching issue on the newsblur server end maybe (just a guess on my part) because the hnrss feed I had on my newsblur account was pulling updates correctly but the search was not working. I tried deleting and re-adding the feed to fix the search (which does now work) but now it does not pull the latest updates from hacker news. Going to the hnrss.org website and manually looking at their rss feed for a filter with 75 points does show properly updating stories.

**Comment by rattasak** [comment]
I did a little more poking around, and what I've noticed is that adding the feed initially finds 10 stories and search works. If I go into "Site Settings" and manually click the button to "Parse this XML/RSS feed" I get all the stories correctly but then search stops working (no results returned for any search terms). The feed address I'm using is http://hnrss.org/newest?points=75

**Comment by rattasak** [comment]
Another piece of info that might be helpful in debugging. When I create a feed from the same site but with points set to 50 or 100, the number of subscribers is 43 for both, but for the feed with points set to 75, the number of subscribers is -1. Searching works for the feeds with points filters set to 50 or 100 and the initial pull from the feed is 500 stories (versus 10 for the points filter set to 75).

**Comment by rattasak** [comment]
Ah, didn't realize that the getsatisfaction site is the place for issues. Closing this one for now.

**Comment by samuelclay** [comment]
Thanks for writing this up. I fixed this in 8a3563766fb2124f5f1f394337610f4f19675d7b. It should work now. Let me know here if it doesn't. This type of ticket certainly belongs here, as it's a critical bug.

**Comment by rattasak** [comment]
I think it's working properly now. Thanks for the fix! 

Out of curiosity, can you give a quick explanation of the fix? I looked at the changes and it looks like there were only changes to the subscriber counts for a feed, but it fixed the issues with updating and searching I had with the hacker news feed filtered on 75 points.

**Comment by samuelclay** [comment]
Yeah, it's half and half. Your feed had what I call a "branched feed", which is when somebody changes the feed address of a feed. Instead of creating a new feed, I branch the feed so that it retains its connections with the original. When subscriber counts are done, branched feeds count towards the subscriber count, so you aren't penalized for branching a feed. In this case, something went wrong and the branched feed had the correct address and the original feed had an incorrect reference to the address. 

This caused a conflict, which I manually resolved. I used to automatically resolve these types of conflicts, but I stopped doing so because some feeds were being eaten up, and these were feeds with tens of thousands of subscribers being automatically merged into a branched feed just because the original feed had a temporary issue.
